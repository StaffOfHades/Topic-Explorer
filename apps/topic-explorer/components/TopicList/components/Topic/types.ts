import { Topic_topic_relatedTopics as Topic } from '../../../../graphql/Topic';

export interface TopicProps {
  className?: string;
  topic: Omit<Topic, '__typename'>;
  setTopic(topic: string): void;
}
