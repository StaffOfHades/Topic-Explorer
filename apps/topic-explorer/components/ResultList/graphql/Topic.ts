/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: Topic
// ====================================================

export interface Topic_topic_relatedTopics {
  __typename: "Topic";
  id: string;
  /**
   * The topic's name.
   */
  name: string;
  /**
   * Returns a count of how many stargazers there are on this object
   */
  stargazerCount: number;
}

export interface Topic_topic {
  __typename: "Topic";
  /**
   * A list of related topics, including aliases of this topic, sorted with the most relevant
   * first. Returns up to 10 Topics.
   */
  relatedTopics: Topic_topic_relatedTopics[];
}

export interface Topic {
  /**
   * Look up a topic by name.
   */
  topic: Topic_topic | null;
}

export interface TopicVariables {
  name: string;
}
