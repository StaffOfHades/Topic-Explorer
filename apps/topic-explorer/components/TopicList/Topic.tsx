import React from 'react';

import { TopicProps } from './types';

export const Topic = ({ className, topic, setTopic }: TopicProps) => {
  const handler = () => {
    setTopic(topic.name);
  };
  return (
    <li className={className}>
      <button onClick={handler}>
        {topic.name}
        <span>â˜… {topic.stargazerCount}</span>
      </button>
    </li>
  );
};
