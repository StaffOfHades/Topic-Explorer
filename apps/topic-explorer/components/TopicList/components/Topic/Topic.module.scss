.topic {
  span:first-child {
    flex-grow: 4;
  }
}
