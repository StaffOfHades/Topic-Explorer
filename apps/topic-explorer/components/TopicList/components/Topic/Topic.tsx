import React from 'react';
import classNames from 'classnames';

import { TopicProps } from './types';
import styles from './Topic.module.scss';

export const Topic = ({
  className,
  topic: { name, stargazerCount },
  setTopic,
}: TopicProps) => {
  const onClickHandler = () => {
    setTopic(name);
  };

  const buttonClassNames = [
    'flex',
    'w-full',
    'px-4',
    'py-1',
    'border-2',
    'border-gray-300',
    'shadow',
    'rounded-lg',
    'active:bg-gray-400',
    'hover:border-gray-500',
    'focus:ring-2',
    'focus:ring-blue-400',
  ].join(' ');

  return (
    <li className={classNames(className, styles.topic, 'block my-2')}>
      <button className={buttonClassNames} onClick={onClickHandler}>
        <span className="inline-block flex-auto text-left text-lg">{name}</span>
        <span className="inline-block flex-auto text-right text-yellow-600">
          â˜… {stargazerCount}
        </span>
      </button>
    </li>
  );
};
