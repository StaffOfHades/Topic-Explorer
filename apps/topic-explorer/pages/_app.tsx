import { AppProps } from 'next/app';
import {
  ApolloClient,
  ApolloProvider,
  InMemoryCache,
  createHttpLink,
} from '@apollo/client';
import { setContext } from '@apollo/client/link/context';
import Head from 'next/head';
import 'tailwindcss/tailwind.css';

const httpLink = createHttpLink({
  uri: 'https://api.github.com/graphql',
});

const authLink = setContext((_, { headers }) => ({
  headers: {
    ...headers,
    authorization: `Bearer ${process.env.NEXT_PUBLIC_GITHUB_API_TOKEN}`,
  },
}));

const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache(),
});

function CustomApp({ Component, pageProps }: AppProps) {
  return (
    <ApolloProvider client={client}>
      <Head>
        <title>GitHub Topic Explorer</title>
      </Head>
      <div className="app">
        <Component {...pageProps} />
      </div>
    </ApolloProvider>
  );
}

export default CustomApp;
