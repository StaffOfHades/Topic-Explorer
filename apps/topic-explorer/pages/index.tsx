import { useState } from 'react';
import { useQuery, gql } from '@apollo/client';

import styles from './index.module.scss';
import { TopicList } from '../components/TopicList';
import { Topic as TopicData, TopicVariables } from '../graphql/Topic';

const SEARCH_TOPICS = gql`
  query Topic($name: String!) {
    topic(name: $name) {
      relatedTopics(first: 10) {
        id
        name
        stargazerCount
      }
    }
  }
`;

export function Index() {
  const [topic, setTopic] = useState('react');
  const { data, loading } = useQuery<TopicData, TopicVariables>(SEARCH_TOPICS, {
    variables: { name: topic },
  });
  return (
    <TopicList
      loading={loading}
      topic={topic}
      topics={data?.topic?.relatedTopics ?? []}
      setTopic={setTopic}
    />
  );
}

export default Index;
