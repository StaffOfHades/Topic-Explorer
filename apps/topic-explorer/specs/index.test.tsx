import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { act, create } from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';

import { Index, FIND_TOPIC_QUERY } from '../pages/index';
import { TopicVariables } from '../graphql/Topic';
import topicData from '../fixtures/Topic.json';

describe('/', () => {
  const findTopicVariable: TopicVariables = { name: 'topicName' };
  const findTopicMock = {
    request: {
      query: FIND_TOPIC_QUERY,
      variables: findTopicVariable,
    },
    result: topicData,
  };

  it('renders index page unchanged', async () => {
    const component = create(
      <MockedProvider addTypename={false} mocks={[findTopicMock]}>
        <Index initialTopic={findTopicVariable.name} />
      </MockedProvider>
    );
    await act(
      async () =>
        await waitFor(() =>
          expect(component.root.findAllByType('button')).toHaveLength(
            topicData.data.topic.relatedTopics.length
          )
        )
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
