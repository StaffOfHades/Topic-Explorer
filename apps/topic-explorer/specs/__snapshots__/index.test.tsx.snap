// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ renders index page unchanged 1`] = `
<div
  className="container mx-auto text-center"
>
  <h1
    className="text-3xl font-medium px-2 py-4"
  >
    topicName
  </h1>
  <hr />
  <ul
    className="topicsList mx-auto px-2 my-4"
  >
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          angular
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          35411
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          react-native
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          20032
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          vue
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          41685
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          opencv
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          921
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          computer-vision
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          1564
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          object-detection
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          150
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          deep-learning
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          41886
        </span>
      </button>
    </li>
    <li
      className="topic block my-2"
    >
      <button
        className="flex w-full px-4 py-1 border-2 border-gray-300 shadow rounded-lg active:bg-gray-400 hover:border-gray-500 focus:ring-2 focus:ring-blue-400"
        onClick={[Function]}
      >
        <span
          className="inline-block flex-auto text-left text-lg"
        >
          tensorflow
        </span>
        <span
          className="inline-block flex-auto text-right text-yellow-600"
        >
          ★ 
          24431
        </span>
      </button>
    </li>
  </ul>
</div>
`;
